Codeinwp/otter-blocks
# Add AI Content

Closes #.


### Summary


Add content generation feature in AI Block.

Now Core Paragraph and Heading are enhanced using AI Toolbar, which provides useful premade prompts for text generation. This feature is also available when you multi-select multiple paragraphs and headings.

ℹ️ AI Block generates the content inside itself and provides features like history. AI Toolbar inserts the content **directly** into the page.

Additional:
- Added section to change the OpenAI Key in Dashboard > Integration
- Solved the pesky behavior of not getting the request when they are retrieved from an external server
- Now, the generation request goes through a proxy where we measure the usage. 

### Screenshots 

#### AI Block Generation

![image](https://github.com/Codeinwp/otter-blocks/assets/17597852/26105a45-fd57-42c7-bfa5-6f2077e80caf)
![image](https://github.com/Codeinwp/otter-blocks/assets/17597852/95ff8c8e-1d21-4c7d-8849-0517a0a4d9da)

#### AI Toolbar

![image](https://github.com/Codeinwp/otter-blocks/assets/17597852/d97fd415-a4c3-4b3a-9edf-fddf92dbc4cc)

https://github.com/Codeinwp/otter-blocks/assets/17597852/0822fd07-54ad-4af5-900a-e2e5480b0973




----

### Test instructions


⚠️ You will need an Open AI API Key. 

- Test the new feature of the AI Block: Content generation. Make sure that From generation works the same.
- Test the AI Toolbar for Heading and Paragraph. Check for glitches.



---- 

### Checklist before the final review
